django/db/models/sql/query.py

```diff
@@ -890,18 +897,25 @@ def join(self, join, reuse=None):
         The 'reuse' parameter can be either None which means all joins are
         reusable, or it can be a set containing the aliases that can be reused.
 
+        The 'force_reuse' parameter is the same type of 'reuse' but with difference
+        that it bypasses the 'reuse' paramter. It is used when computing.
+
         A join is always created as LOUTER if the lhs alias is LOUTER to make
         sure chains like t1 LOUTER t2 INNER t3 aren't generated. All new
         joins are created as LOUTER if the join is nullable.
         """
-        reuse = [a for a, j in self.alias_map.items()
-                 if (reuse is None or a in reuse) and j == join]
+        if force_reuse:
+            reuse = [a for a, j in self.alias_map.items()
```

Comment:
Do you mean "Use four space hanging indentation rather than vertical alignment" ?

```diff
diff --git i/django/db/models/sql/query.py w/django/db/models/sql/query.py
index 3486a4c667..e3ac77bb00 100644
--- i/django/db/models/sql/query.py
+++ w/django/db/models/sql/query.py
@@ -905,8 +905,10 @@ def join(self, join, reuse=None, force_reuse=None):
         joins are created as LOUTER if the join is nullable.
         """
         if force_reuse:
-            reuse = [a for a, j in self.alias_map.items()
-                     if a in force_reuse and j.equals(join, with_filtered_relation=False)]
+            reuse = [
+                a for a, j in self.alias_map.items()
+                if a in force_reuse and j.equals(join, with_filtered_relation=False)
+            ]
         else:
             reuse = [a for a, j in self.alias_map.items()
                      if (reuse is None or a in reuse) and j == join]
```

Really?
It makes a bit of sense in the example at least, where the message is more likely to change - but Python uses to be more for the readers than the writers IIRC.

Reply:
Would it be OK like this?

```diff
diff --git c/django/db/models/sql/query.py i/django/db/models/sql/query.py
index 3486a4c667..c83bd9dd5b 100644
--- c/django/db/models/sql/query.py
+++ i/django/db/models/sql/query.py
@@ -905,8 +905,9 @@ def join(self, join, reuse=None, force_reuse=None):
         joins are created as LOUTER if the join is nullable.
         """
         if force_reuse:
-            reuse = [a for a, j in self.alias_map.items()
-                     if a in force_reuse and j.equals(join, with_filtered_relation=False)]
+            reuse = [
+                a for a, j in self.alias_map.items()
+                if a in force_reuse and j.equals(join, with_filtered_relation=False)]
         else:
             reuse = [a for a, j in self.alias_map.items()
                      if (reuse is None or a in reuse) and j == join]
@@ -1159,8 +1160,9 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
         allow_many = not branch_negated or not split_subq
 
         try:
-            join_info = self.setup_joins(parts, opts, alias, can_reuse=can_reuse, force_reuse=force_reuse,
-                                         allow_many=allow_many)
+            join_info = self.setup_joins(
+                parts, opts, alias, can_reuse=can_reuse, force_reuse=force_reuse,
+                allow_many=allow_many)
 
             # Prevent iterator from being consumed by check_related_objects()
             if isinstance(value, Iterator):
```

Reply:
Put the `]` on a newline, like the style guide says.